src/key.rs

```diff
@@ -268,54 +267,87 @@ where
     values: Vec<Option<RedisString>>,
 }
 
-impl<'a, T> HMGetResult<'a, T>
+pub struct HMGetIter<'a, T>
+where
+    T: Into<Vec<u8>>,
+{
+    ai: std::slice::Iter<'a, T>,
+    bi: std::vec::IntoIter<Option<RedisString>>,
+}
+
+impl<'a, T> Iterator for HMGetIter<'a, T>
 where
     T: Into<Vec<u8>> + Clone,
 {
-    /// Convert the results into any list-like collection. The field values in the collection
-    /// appear in the same order as that in which the field names were provided to the multi-get
-    /// call. Fields which do not exist are set to `None`.
+    type Item = (T, String);
```

Comment:
Using a `String` for the value is limiting -- what if the value is not a valid UTF-8 string? In any case, the iteration and the string conversion seem to be separate concerns. Why not let the caller use `map` with whatever conversion they prefer? Also, it might be possible to use `Into` and then call `into()` on the value to let the caller specify the conversion.

Reply:
Point taken. I've pushed a commit that makes this more flexible. You can now iterate by any type that satisfies `RedisString: Into<T>`. (And added a corresponding conversion for `String` for convenience.)